### PR TITLE
[WIP] Update initialization of boundary conditions

### PR DESCRIPTION
**Pull request by RemiLehe**
On Cori, we saw that the logic of setting the static members of WarpX for boundary conditions causes undefined behavior if we set those members before we instantiate a WarpX object itself.

We fix this by splitting the logic of overwriting AMReX periodicity in the param parser and the actual WarpX member init.

- [x] Cori test (C++): Successful
- [x] Update PICMI/Python

### CODE REVIEW

**Comment by ax3l** [comment]
We still need to update the PICMI setup logic.
With the Python tests:
```
1::Assertion `geom_periodicity[idim] == 1' failed, file "/Users/runner/work/WarpX/WarpX/Source/Utils/WarpXUtil.cpp", line 683, Msg: "amrex boundary condition (as set in function OverwriteAMReXperiodic) is inconsistent." !!!
```

Update: I pushed a commit.

**Review by ax3l — APPROVED** [review]

File: Source/Utils/WarpXUtil.cpp [hunk 1]
```diff
@@ -602,6 +602,34 @@ void CheckGriddingForRZSpectral ()
...
+    amrex::Vector<int> geom_periodicity(AMREX_SPACEDIM,0);
+    ParmParse pp_geometry("geometry");
+    if (pp_geometry.queryarr("is_periodic", geom_periodicity)) {
+        amrex::Abort("geometry.is_periodic is not supported. Please use `boundary.field_lo`, `boundary.field_hi` to specifiy field boundary conditions and 'boundary.particle_lo', 'boundary.particle_hi'  to specify particle boundary conditions.");
+    }
+
+    // Check which dimention of space is periodic, according to
+    // `boundary.field_lo`. (Consistency between `field_lo` and `field_hi`
+    // is checked in the function `ReadBCParams`)
+    amrex::Vector<std::string> field_BC_lo(AMREX_SPACEDIM,"default");
```
Comment (dpgrote) [inline Source/Utils/WarpXUtil.cpp:617]: A tiny detail - the might be easier to follow if the variable name is the same as the input parameter name, `field_lo`.
